compiler/packages/babel-plugin-react-compiler/src/HIR/CollectHoistablePropertyLoads.ts

```diff
@@ -252,6 +278,18 @@ function collectNonNullsInBlocks(
     const assumedNonNullObjects = new Set<PropertyPathNode>(
       knownNonNullIdentifiers,
     );
+
+    nodes.set(block.id, {
+      block,
+      assumedNonNullObjects,
+    });
+    const maybeOptionalChain = hoistableFromOptionals.get(block.id);
+    if (maybeOptionalChain != null) {
```

Comment:
This is already computed by `collectOptionalChainDependencies` to be non-optional loads (i.e. `a?.b` in `a?.b.c`), so changes to core logic here are minimal